build.gradle

```diff
@@ -901,6 +901,44 @@ project(':server') {
   }
 }
 
+project(':share') {
+  base {
+    archivesName = "kafka-share"
```

Comment:
I think this name is too generic, perhaps `kafka-share-common` would be more aligned with the existing project convention.

That said, do we expect this module to have a reasonable set of classes? If not, a dedicated module may not be worth it.

When it comes to modules, it's important to strike a balance. Too many modules make the system harder to maintain and understand. And they also get exposed to users since these end up as separate jars, Maven modules, etc. Similarly, the opposite results in bad compilation and test time, etc.

Reply:
@ijuma Thanks for the feedback. 

> That said, do we expect this module to have a reasonable set of classes?

Though initally I was of an opinion to have code in `server` module itself but then I agree with @mumrah that it might be sensible to keep in separate module as the code is growing for share groups.

> I think this name is too generic, perhaps kafka-share-common

Sure, do you think `share-group` would be a better name? As module currently holds code related to `share-fetch` and `share-persister`, which are part of overall `share-groups` to function.